### PR TITLE
Add owasp dependency check analysis

### PR DESCRIPTION
**Pull request by Pavel-Lapatin**
Add scrips to review if dependencies are vulnerable or not

### CODE REVIEW

File: src/dotnet-analysis.ps1 [hunk 1]
```diff
@@ -28,4 +28,11 @@ Write-Host "Running roslyn..."
 $roslynOutputPath = "$OutputDirectoryPath\roslyn"
 powershell "$PSScriptRoot\roslyn\run.ps1 -SolutionDirectoryPath $SolutionDirectoryPath -OutputDirectoryPath $roslynOutputPath -SolutionRelativePath $SolutionRelativePath"
 
-Write-Host "Roslyn analysis completed."
\ No newline at end of file
+Write-Host "Roslyn analysis completed."
+
+Write-Host "Running owasp dependency check..."
+
+$clocOutputPath = "$OutputDirectoryPath\security"
```
Comment (pNovikau) [inline src/dotnet-analysis.ps1]:
```suggestion
$securityOutputPath = "$OutputDirectoryPath\security"
```

File: src/dotnet-analysis.sh [hunk 2]
```diff
@@ -46,4 +46,11 @@ echo "Running roslyn..."
 roslynOutputPath="outputDirectoryPath/roslyn"
 sh "$scriptRoot"/roslyn/run.sh -solutionDirectoryPath "$solutionDirectoryPath" -outputDirectoryPath "$roslynOutputPath" -solutionRelativePath "$solutionRelativePath"
 
-echo "Roslyn analysis completed."
\ No newline at end of file
+echo "Roslyn analysis completed."
+
+echo "Running owasp dependency check..."
+
+roslynOutputPath="outputDirectoryPath/security"
```
Comment (pNovikau) [inline src/dotnet-analysis.sh]:
```suggestion
securityOutputPath="outputDirectoryPath/security"
```

File: src/security/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+| SolutionDirectoryPath | True     | String | Path to the solution folder        |
+| SolutionRelativePath  | True     | String | Relative path to the solution file |
+| Output                | True     | String | Path to the output directory       |
+
+## Example
+
+- Clone [CodeReview.AdHocReview](https://github.com/GodelTech/CodeReview.AdHocReview)
+- Clone [CodeReview.Orchestrator](https://github.com/GodelTech/CodeReview.Orchestrator) (can be any project you want to analyze)
+- Navigate to `CodeReview.AdHockCheck\src\security\` folder
+- Run `.\run.ps1 -SolutionDirectoryPath C:\CodeReview.Orchestrator -SolutionRelativePath CodeReview.Orchestrator.sln -Output C:\output`
```
Comment (pNovikau) [inline src/security/README.md]:
```suggestion
- [Windows] Run `.\run.ps1 -SolutionDirectoryPath C:\CodeReview.Orchestrator -SolutionRelativePath CodeReview.Orchestrator.sln -Output C:\output`
- [Linux] Run .\run.sh -SolutionDirectoryPath /CodeReview.Orchestrator -SolutionRelativePath CodeReview.Orchestrator.sln -Output /output
```

File: src/security/run.ps1 [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+IF (-not $?) { exit }
+
+$tmpFile = New-TemporaryFile
+
+Write-Host "Running analysis for the owasp dependency check workflow..."
+
+Copy-Item -Path $PSScriptRoot\workflow.yaml  -Destination $tmpFile
+
+$tmpFileContent = Get-Content -Path $tmpFile
+$tmpFileContent = $tmpFileContent.Replace('{SOLUTION_FILE_PATH}', $SolutionRelativePath)
```
Comment (pNovikau) [inline src/security/run.ps1]: you can remove this line. It was used to replace the `{SOLUTION_FILE_PATH}` with the relative solution path in the workflow file.

File: src/security/run.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+echo "Docker pulling image godeltech/codereview.orchestrator..."
+
+if ! docker image pull godeltech/codereview.orchestrator; then
+  exit
+fi
+
+tmpfile=$(mktemp)
+scriptRoot=$(dirname "$0")
+
+awk "{sub(\"{SOLUTION_FILE_PATH}\",\"$solutionRelativePath\")}1" "$scriptRoot/workflow.yaml" > "$tmpfile"
```
Comment (pNovikau) [inline src/security/run.sh]: you can remove this line. It was used to replace the `{SOLUTION_FILE_PATH}` with the relative solution path in the workflow file.

Comment (pNovikau) [reply src/security/run.sh]: Also, you can remove `SolutionRelativePath` from the parameters of the script as well as from the README file

**Review by pNovikau — APPROVED** [review]

**Review by pNovikau — CHANGES_REQUESTED** [review]
Please see the comments above

File: src/security/run.sh (hunk 5 above)
Comment (Pavel-Lapatin) [reply src/security/run.sh]: Done

**Review by pNovikau — APPROVED** [review]

**Pull request merged**